dask/templates/additional-worker-deployment.yaml

```diff
@@ -0,0 +1,97 @@
+{{- range $worker := .Values.additional_worker_groups }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "dask.fullname" . }}-worker
+  labels:
+    app: {{ template "dask.name" . }}
+    heritage: {{ $.Release.Service | quote }}
+    release: {{ $.Release.Name | quote }}
+    chart: {{ template "dask.chart" . }}
+    component: worker
+spec:
+  replicas: {{ if $worker.replicas }}{{ $worker.replicas }}{{ else }}{{ $.Values.worker.replicas }}{{ end }}
+  selector:
+    matchLabels:
+      app: {{ template "dask.name" . }}
+      release: {{ $.Release.Name | quote }}
+      component: worker
+  strategy:
+    type: {{ $worker.strategy.type }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "dask.name" . }}
+        release: {{ $.Release.Name | quote }}
+        component: worker
+      {{- if $worker.annotations }}
+      annotations:
+        {{- toYaml $worker.annotations | nindent 8 }}
+      {{- end }}
+    spec:
+      imagePullSecrets:
+        {{- toYaml $worker.image.pullSecrets | nindent 8 }}
+      {{- if $worker.mounts.volumes }}
+      volumes:
+        {{- toYaml $worker.mounts.volumes | nindent 8}}
+      {{- end }}
+      containers:
+        - name: {{ template "dask.fullname" . }}-worker
+          image: "{{ $worker.image.repository }}:{{ $worker.image.tag }}"
```

Comment:
@jacobtomlinson Should I check both repository and tag and use the default values if either one is missing?

Reply:
No I would treat them separately.